### PR TITLE
 Improved monitoring of replica state during large binlog load

### PR DESCRIPTION
**Pull request by WithSoull**
# Improved monitoring of replica state during large binlog load

## Description:
This pull request addresses an issue where the ZooKeeper service mistakenly believed that a replica was dead when it was actually downloading a large binlog file. 

To resolve this problem, I added a new field to the replica — NodeState.IsBinlogLoading. This field will allow us to more accurately track the state of the replica and ensure that the service correctly identifies when a replica is in the process of downloading a binlog.

Additionally, I have written unit tests and integration tests that cover the new functionality I have added.

## Small fixes
I have also made minor updates to the existing code in the unit tests to improve the readability of the tests.

Finally, I have updated the Dockerfile to reflect the changes made in this pull request. This update removes two warnings and ensures that the latest version of the code is being used in the container.

### CODE REVIEW

File: internal/app/data.go [hunk 1]
```diff
@@ -144,6 +145,20 @@ func (ns *NodeState) CalcGTIDDiffWithMaster() (string, error) {
 	return gtids.GTIDDiff(replicaGTID, sourceGTID)
 }
 
+func (ns *NodeState) UpdateBinlogStatus(oldBinloPos string) (newBinlogPos string) {
+	if ns.SlaveState != nil {
+		newBinlogPos = fmt.Sprintf("%s%019d", ns.SlaveState.MasterLogFile, ns.SlaveState.MasterLogPos)
```
Comment (teem0n) [inline internal/app/data.go:150]: Let's extract this to separate function

File: internal/app/data.go [hunk 2]
```diff
@@ -144,6 +145,20 @@ func (ns *NodeState) CalcGTIDDiffWithMaster() (string, error) {
...
+
+		if newBinlogPos > oldBinloPos {
```
Comment (teem0n) [inline internal/app/data.go:152]: ns.IsLoadingBinlog = newBinlogPos > oldBinloPos

**Review by teem0n — APPROVED** [review]

**Review by noname0443 — APPROVED** [review]

**Review by noname0443 — APPROVED** [review]

**Comment by noname0443** [comment]
I believe it is a good idea to aggregate tests into two scripts, because it reduces consuming resources on creating a container on each test, but I suppose we may have a problem in the future with flapped tests. If one is dead due to context deadline, we have to restart a whole test set for either 5.7 or 8.0.

I think we should separate CI/CD changes in a separate PR and maybe add some restart logic for flapped tests.

**Comment by WithSoull** [comment]
Yes, I think you are right, we should split the PR.

I want to add my thoughts on this.
We need to set some kind of restart limit so that it doesn't go on indefinitely.

**Comment by noname0443** [comment]
BTW: In the current view, we launch tests in a sequence. So it can be a good idea to launch them in a parallel way.

**Review by noname0443 — APPROVED** [review]

**Review by noname0443 — APPROVED** [review]

File: tests/features/repl_mon.feature [hunk 3]
```diff
@@ -51,3 +70,22 @@ Feature: repl_mon tests
...
+
+    Then zookeeper node "/test/health/mysql2" should match json within "20" seconds
+      """
+      {
+        "is_loading_binlog": false
+      }
+      """
+
+    And mysql host "mysql3" should be replica of "mysql1"
+
```
Comment (noname0443) [inline tests/features/repl_mon.feature:84]: Why should we leave a gap between lines?

**Review by noname0443 — APPROVED** [review]

**Pull request merged**